### PR TITLE
Add NOI Documents

### PR DESCRIPTION
**Pull request by dhaselhan**
* Add new DB table
* Share DocumentType across App and NOI Docs
* Add new tab to detailed page

### CODE REVIEW

File: services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-document/notice-of-intent-document.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+  it('should call through for fetchTypes', async () => {
+    mockTypeRepository.find.mockResolvedValue([]);
+
+    const res = await service.fetchTypes();
+
+    expect(mockTypeRepository.find).toHaveBeenCalledTimes(1);
+    expect(res).toBeDefined();
+  });
+
+  // it('should set the type and description for multiple files', async () => {
```
Comment (mhuseinov) [inline services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-document/notice-of-intent-document.service.spec.ts:251]: I assume this one is also going to be enabled later?

File: services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-document/notice-of-intent-document.service.ts [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+      relations: this.DEFAULT_RELATIONS,
+    });
+    if (!document) {
+      throw new NotFoundException(`Failed to find document ${uuid}`);
+    }
+    return document;
+  }
+
+  async delete(document: NoticeOfIntentDocument) {
+    await this.noticeOfIntentDocumentRepository.remove(document);
```
Comment (mhuseinov) [inline services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-document/notice-of-intent-document.service.ts:142]: completely optional, now we know that this one could be wrapped into transaction in typeorm

File: services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-meeting/notice-of-intent-meeting.controller.ts [hunk 3]
```diff
@@ -83,7 +83,7 @@ export class NoticeOfIntentMeetingController {
     fileNumber: string,
     meeting: CreateNoticeOfIntentMeetingDto,
   ) {
-    const noi = await this.noiService.getOrFail(fileNumber);
+    const noi = await this.noiService.getOrFailByUuid(fileNumber);
```
Comment (mhuseinov) [inline services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-meeting/notice-of-intent-meeting.controller.ts:86]: a little bit odd to have fileNumber as paramter for getOrFailByUuid but I assume this is a leftover from "application"

**Review by mhuseinov — APPROVED** [review]
I like that Document code is shared between application and noi 👍

File: services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-document/notice-of-intent-document.service.spec.ts (hunk 1 above)
Comment (dhaselhan) [reply services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-document/notice-of-intent-document.service.spec.ts:252]: yes, added a comment specifying this is for Step 7

File: services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-meeting/notice-of-intent-meeting.controller.ts (hunk 3 above)
Comment (dhaselhan) [reply services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-meeting/notice-of-intent-meeting.controller.ts]: renamed to uuid as it is uuid

**Pull request merged**